cvmfs/ssl.cc

```diff
@@ -0,0 +1,109 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_SSL_H_
+#define CVMFS_SSL_H_
+
+#include "ssl.h"
+
+#include <cstdlib>
+#include <string>
+#include <vector>
+
+#include "duplex_curl.h"
+#include "platform.h"
+#include "util/posix.h"
+#include "util/string.h"
+
+#ifdef CVMFS_NAMESPACE_GUARD
```

Comment:
The `CVMFS_NAMESPACE_GUARD` is defined only when building the loader.  The idea is to use the same code in the loader (cvmfs2 binary) and in the libcvmfs_fuse library without name clashes.

Here, we would simply open an anonymous namespace containing everything except the public `AddSSLCertificates` function